source/mesh_deformation/interface.cc

```diff
@@ -722,6 +740,253 @@ namespace aspect
 
 
 
+    template <int dim>
+    void MeshDeformationHandler<dim>::compute_mesh_displacements_gmg()
+    {
+      // Same as compute_mesh_displacements, but using GMG instead of AMG.
+
+      // We use this gmg solver only when the gmg stokes solver is used
+      // for the following reasons (TODO):
+      // 1. this gmg solver does not support periodic boundary conditions
+      // 2. To use this solver even when gmg stokes solver is not used, we need to
+      //    initialize the triangulation with Triangulation<dim>::limit_level_difference_at_vertices
+      //    and parallel::distributed::Triangulation<dim>::construct_multigrid_hierarchy
+      // 3. Although this gmg solver is much faster than the amg solver, it's only tested for
+      //    limited free surface cases.
+
+      TimerOutput::Scope timer (this->sim.computing_timer, "gmg displacement");
+
+      Assert(mesh_deformation_fe.degree == 1, ExcNotImplemented());
+      const unsigned int mesh_deformation_fe_degree = 1;
```

Comment:
Wouldnt this be more consistent?
```suggestion
      const unsigned int mesh_deformation_fe_degree = mesh_deformation_fe.degree;
```

Reply:
Yes, it needs to be known at compile time. I don't think it is a big issue because we don't support higher order displacements anyways at this point.